src/main/java/net/minecraftforge/fml/common/network/NetworkRegistry.java

```diff
@@ -304,13 +306,18 @@ public void register(ModContainer fmlModContainer, Class<?> clazz, @Nullable Str
 
     public boolean isVanillaAccepted(Side from)
     {
-        boolean result = true;
-        for (Entry<ModContainer, NetworkModHolder> e : registry.entrySet())
-        {
-            result &= e.getValue().acceptsVanilla(from);
-        }
-        return result;
+        return registry.values().stream()
+                .allMatch(mod -> mod.acceptsVanilla(from));
     }
+
+    public Collection<String> getModNamesThatDoNotAcceptVanilla(Side from)
+    {
+        return registry.values().stream()
```

Comment:
Might be nice to have the list here sorted?

Reply:
Yeah, it's not performance-critical or anything so might as well make it nice. Will do.